frontends/common/constantFolding.cpp

```diff
@@ -220,10 +220,12 @@ const IR::Node* DoConstantFolding::preorder(IR::ArrayIndex* e) {
                 return e;
             }
             auto value = getConstant(e->left);
+            if (!value)
```

Comment:
What causes this expression to be null in the first place?

Reply:
This returns null if the expression is not a constant.
For an array it will return a non-null constant only if the array is a list expression.
When we add array initializers we'll be able to have constant arrays too (I am not sure that having constant headers is useful, though, but it should work).